shared/actions/profile.js

```diff
@@ -185,13 +320,83 @@ function submitRevokeProof (proofId: string): AsyncAction {
   }
 }
 
+function checkProof (): AsyncAction {
+  return (dispatch, getState) => {
+    dispatch(updateError(null))
+    const sigID = getState().profile.sigID
+
+    // The initial checkProof() call happens while we're in the middle of the incomingCallMap flow. That can error out
+    // and we can call this again. This if case is when we're in the incomingCallMap
+    if (outputInstructionsResponse) {
+      outputInstructionsResponse.result()
+      outputInstructionsResponse = null
+
+      return // the proveStartProofRpc will call checkProof after we respond to outputInstructionsResponse automatically
+    }
+
+    proveCheckProofRpc({
+      ...makeWaitingHandler(dispatch),
+      param: {
+        sigID,
+      },
+      callback: (error, {found, status}) => {
+        if (error) {
+          console.warn('Error getting proof update')
+          dispatch(updateError("We couldn't verify your proof. Please retry!"))
+        } else {
+          // this enum value is the divider between soft and hard errors
+          if (!found && status >= proveCommon.ProofStatus.baseHardError) {
+            dispatch(updateError("We couldn't find your proof. Please retry!"))
+          } else {
+            dispatch(updateProofStatus(found, status))
+            dispatch(navigateTo([{path: 'ConfirmOrPending'}], profileTab))
+          }
+        }
+      },
+    })
+  }
+}
+
+function outputInstructionsActionLink (): AsyncAction {
+  return (dispatch, getState) => {
+    const profile = getState().profile
+    switch (profile.platform) {
+      case 'coinbase':
+        shell.openExternal(`https://coinbase.com/${profile.username}#settings`)
+        break
+      case 'twitter':
+        shell.openExternal(`https://twitter.com/home?status=${profile.proof}`)
+        break
+      case 'github':
+        shell.openExternal('https://gist.github.com/')
+        break
+      case 'reddit':
+        shell.openExternal(profile.proof)
+        break
+      default:
+        break
+    }
+  }
+}
+
+function reloadProfile (): AsyncAction {
+  return (dispatch) => {
+    dispatch(getMyProfile())
+    dispatch(navigateUp())
```

Comment:
cool! that makes sense, maybe a different name though? how about `reloadProfileAndNavUp`